api/types/types.proto

```diff
@@ -2704,41 +2704,78 @@ message SSODiagnosticInfo {
+    SSOWarnings SAMLAttributesToRolesWarnings = 11
         [ (gogoproto.jsontag) = "saml_attributes_to_roles_warnings,omitempty" ];
 
     // SAMLAttributeStatements represents SAML attribute statements.
-    wrappers.LabelValues SAMLAttributeStatements = 7 [
+    wrappers.LabelValues SAMLAttributeStatements = 12 [
         (gogoproto.nullable) = false,
         (gogoproto.jsontag) = "saml_attribute_statements,omitempty",
         (gogoproto.customtype) = "github.com/gravitational/teleport/api/types/wrappers.Traits"
     ];
 
     // SAMLAssertionInfo represents raw SAML assertion info as returned by IdP during SAML flow.
-    bytes SAMLAssertionInfo = 8 [
+    bytes SAMLAssertionInfo = 13 [
         (gogoproto.jsontag) = "saml_assertion_info,omitempty",
         (gogoproto.customtype) = "AssertionInfo"
     ];
 
     // SAMLTraitsFromAssertions represents traits translated from SAML assertions.
-    wrappers.LabelValues SAMLTraitsFromAssertions = 9 [
+    wrappers.LabelValues SAMLTraitsFromAssertions = 14 [
         (gogoproto.nullable) = false,
         (gogoproto.jsontag) = "saml_traits_from_assertions,omitempty",
         (gogoproto.customtype) = "github.com/gravitational/teleport/api/types/wrappers.Traits"
     ];
 
     // SAMLConnectorTraitMapping represents connector-specific trait mapping.
-    repeated TraitMapping SAMLConnectorTraitMapping = 10 [
+    repeated TraitMapping SAMLConnectorTraitMapping = 15 [
         (gogoproto.nullable) = false,
         (gogoproto.jsontag) = "saml_connector_trait_mapping,omitempty"
     ];
+
+    // OIDCClaimsToRoles specifies a mapping from claims (traits) to teleport roles.
+    repeated ClaimMapping OIDCClaimsToRoles = 20
+        [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "oidc_claims_to_roles,omitempty" ];
+
+    // OIDCClaimsToRolesWarnings contains warnings produced during the process of mapping the
+    // OIDC claims to roles.
+    SSOWarnings OIDCClaimsToRolesWarnings = 21
+        [ (gogoproto.jsontag) = "oidc_claims_to_roles_warnings,omitempty" ];
+
+    // OIDCClaims represents OIDC claims.
+    bytes OIDCClaims = 22 [
+        (gogoproto.nullable) = false,
+        (gogoproto.jsontag) = "oidc_claims,omitempty",
+        (gogoproto.customtype) = "OIDCClaims"
+    ];
+
+    // OIDCIdentity represents mapped OIDC Identity.
+    bytes OIDCIdentity = 23 [
+        (gogoproto.jsontag) = "oidc_identity,omitempty",
+        (gogoproto.customtype) = "OIDCIdentity"
+    ];
+
+    // OIDCTraitsFromClaims represents traits translated from OIDC claims.
+    wrappers.LabelValues OIDCTraitsFromClaims = 24 [
+        (gogoproto.nullable) = false,
+        (gogoproto.jsontag) = "oidc_traits_from_claims,omitempty",
+        (gogoproto.customtype) = "github.com/gravitational/teleport/api/types/wrappers.Traits"
+    ];
+
+    // OIDCConnectorTraitMapping represents connector-specific trait mapping.
+    repeated TraitMapping OIDCConnectorTraitMapping = 25 [
+        (gogoproto.nullable) = false,
```

Comment:
Does nullable gogoproto settings make any sens in case of `repeated` or `bytes` bytes ?

Reply:
Yes, this changes the type of the element. With `nullable = false` we have `[]TraitMapping`, otherwise we would have `[]*TraitMapping`.